lib/imapSimple.js

```diff
@@ -225,9 +261,49 @@ function connect(options, callback) {
     });
 }
 
+/**
+ * Returns a list of the parts.
+ * A part is either a message or an attachment.
+ * Parts are usually nested and numbered things like 1, 1.1, 1.2, ...
+ * Code taken from http://stackoverflow.com/questions/25247207/how-to-read-and-save-attachments-using-node-imap
+ */
+function findParts(struct, parts) {
+    parts = parts ||  [];
+    for (var i = 0; i < struct.length; i++) {
+        if (Array.isArray(struct[i])) {
+	    findParts(struct[i], parts);
+	} else if (struct[i].partID) {
+	    parts.push(struct[i]);
+	}
+    }
+    return parts;
+}
+
+/**
+ * Simplify the structure of the header (subject, date, etc) a bit
+ * Does some simple parsing to pull out useful attributes and all attachments
+ */
+function decorateMessage(message) {
+    if (message.parts.length != 1)
+        throw Error('Message has weird number of parts')
```

Comment:
it is only the case that there is a single part when you request only a single part in your fetchOptions. For example, if I have this is my fetch options (as in the test file `index.spec.js`): 

```js
var fetchOptions = {
    bodies: ['HEADER', 'TEXT'],
    markSeen: false
};
```

You would have to grab the 'HEADER' part from the results. for example:

```js
var header = message.parts.filter(function (part) {
    return part.which === 'HEADER';
})[0];
```